packages/integration_test/ios/Classes/IntegrationTestIosTest.h

```diff
@@ -21,26 +19,48 @@ NS_ASSUME_NONNULL_BEGIN
 
 @end
 
+// For every Flutter dart test, dynamically generate an Objective-C method mirroring the test results
+// so it is reported as a native XCTest run result.
+// If the Flutter dart tests have captured screenshots, add them to the XCTest bundle.
 #define INTEGRATION_TEST_IOS_RUNNER(__test_class)                                           \
-  @interface __test_class : XCTestCase<FLTIntegrationTestScreenshotDelegate>                \
+  @interface __test_class : XCTestCase                                                      \
   @end                                                                                      \
                                                                                             \
   @implementation __test_class                                                              \
                                                                                             \
-  - (void)testIntegrationTest {                                                             \
-    NSString *testResult;                                                                   \
-    IntegrationTestIosTest *integrationTestIosTest = integrationTestIosTest = [[IntegrationTestIosTest alloc] initWithScreenshotDelegate:self]; \
-    BOOL testPass = [integrationTestIosTest testIntegrationTest:&testResult];               \
-    XCTAssertTrue(testPass, @"%@", testResult);                                             \
-  }                                                                                         \
-                                                                                            \
-  - (void)didTakeScreenshot:(UIImage *)screenshot attachmentName:(NSString *)name {         \
-    XCTAttachment *attachment = [XCTAttachment attachmentWithImage:screenshot];             \
-    attachment.lifetime = XCTAttachmentLifetimeKeepAlways;                                  \
-    if (name != nil) {                                                                      \
-      attachment.name = name;                                                               \
+  + (NSArray<NSInvocation *> *)testInvocations {                                            \
```

Comment:
This logic is the same as in https://github.com/flutter/flutter/pull/88013 (except the test selector instead of name part).